src/NHibernate/Engine/Query/QueryPlanCache.cs

```diff
@@ -211,14 +222,7 @@ protected HQLQueryPlanKey(System.Type queryTypeDiscriminator, string query, bool
 					filterNames = new HashSet<string>(enabledFilters.Keys);
 				}
 
-				unchecked
-				{
-					var hash = query.GetHashCode();
-					hash = 29 * hash + (shallow ? 1 : 0);
-					hash = 29 * hash + CollectionHelper.GetHashCode(filterNames, filterNames.Comparer);
```

Comment:
`filterNames.Comparer` is the default one here, the hashset is always constructed by this class, without a custom comparer. The new code does no more use it explicitly, as the hashset may not yet be available, if the hashcode computation has to be done during deserialization.